pkg/verifier/notaryv2/notaryv2.go

```diff
@@ -170,51 +168,61 @@ func (v *notaryV2Verifier) Verify(ctx context.Context,
 	}, nil
 }
 
-// This function is borrowed internals from the notation-go jws verifier
-// https://github.com/notaryproject/notation-go/blob/main/signature/jws/verifier.go
-func getCert(refBlob []byte) (*x509.Certificate, error) {
-	var envelope jwsutil.Envelope
-	if err := json.Unmarshal(refBlob, &envelope); err != nil {
+func getVerifierService(conf *NotaryV2VerifierConfig) (notation.Verifier, error) {
+	policyDoc, err := loadPolicyDocument(conf.TrustPolicy)
+	if err != nil {
 		return nil, err
 	}
-	if len(envelope.Signatures) != 1 {
-		return nil, errors.New("single signature envelope expected")
-	}
-	sig := envelope.Open()
 
-	var header struct {
-		TimeStampToken []byte   `json:"timestamp,omitempty"`
-		CertChain      [][]byte `json:"x5c,omitempty"`
+	store := &trustStore{
+		certPaths: conf.VerificationCerts,
 	}
-	if err := json.Unmarshal(sig.Unprotected, &header); err != nil {
+
+	return notaryVerifier.New(policyDoc, store, nil)
+}
+
+// loadPolicyDocument is brorrowd from notation-go [https://github.com/notaryproject/notation-go/blob/75b3248459d677d2a6abcae3dd880775f4456271/verification/helpers.go#L15]
+func loadPolicyDocument(policyPath string) (*trustpolicy.Document, error) {
+	policyDocument := &trustpolicy.Document{}
+	jsonFile, err := os.ReadFile(policyPath)
+	if err != nil {
 		return nil, err
 	}
-	if len(header.CertChain) == 0 {
-		return nil, errors.New("signer certificates not found")
+
+	if err := json.Unmarshal(jsonFile, &policyDocument); err != nil {
+		return nil, err
 	}
 
-	cert, err := x509.ParseCertificate(header.CertChain[0])
-	if err != nil {
+	if err := policyDocument.Validate(); err != nil {
 		return nil, err
 	}
 
-	return cert, nil
+	return policyDocument, nil
 }
 
-func getVerifierService(certPaths ...string) (notation.Verifier, error) {
-	certs := make([]*x509.Certificate, 0)
-	for _, path := range certPaths {
+func (v *notaryV2Verifier) verifySignature(subjectRef, mediaType string, subjectDesc oci.Descriptor, refBlob []byte) (*notation.VerificationOutcome, error) {
+	opts := notation.VerifyOptions{
+		SignatureMediaType: mediaType,
+		ArtifactReference:  subjectRef,
+	}
 
-		bundledCerts, err := utils.GetCertificatesFromPath(path)
+	return (*v.notationVerifier).Verify(context.Background(), subjectDesc, refBlob, opts)
```

Comment:
can we alter `verifySignature` so that it takes in a `context` that way we can reuse the `context` passed into `Verify`?

Reply:
Good call, I'll make the change.